webapp/crashstats/crashstats/tests/test_jinja_helpers.py

```diff
@@ -329,343 +313,448 @@ def test_with_unicode_signature(self):
         )
 
     def test_comment(self):
-        req = RequestFactory().get("/report/index")
-        raw_crash = {}
-        report = self._create_report()
-        parsed_dump = self._create_dump(
-            threads=[
-                self._create_thread(),  # Empty thread 0
-                self._create_thread(
-                    frames=[
-                        self._create_frame(frame=0),
-                        self._create_frame(frame=1),
-                        self._create_frame(frame=2),
-                    ]
-                ),
-            ]
+        report = self._create_report(
+            crashing_thread=1,
+            json_dump=self._create_dump(
+                threads=[
+                    self._create_thread(),  # Empty thread 0
+                    self._create_thread(
+                        frames=[
+                            self._create_frame(frame=0),
+                            self._create_frame(frame=1),
+                            self._create_frame(frame=2),
+                        ]
+                    ),
+                ]
+            ),
         )
         url = generate_create_bug_url(
-            req, self.TEMPLATE, raw_crash, report, parsed_dump, 1
+            f"http://localhost:8000/report/index/{self.CRASH_ID}",
+            self.TEMPLATE,
+            report,
         )
 
-        # Assert that the crash id is in the comment
-        assert quote_plus(report["uuid"]) in url
-
-        # Assert that the top 3 frames are in the comment
-        assert quote_plus("Top 3 frames of crashing thread:") in url
-
-        frame1 = parsed_dump["threads"][1]["frames"][1]
-        assert (
-            quote_plus("1  {module}  {signature}  {file}:{line}".format(**frame1))
-            in url
+        qs = self._extract_query_string(url)
+        comment = (
+            "Crash report: http://localhost:8000/report/index/70dda764-a402-4ca3-b806-c38dd0240328\n"
+            "\n"
+            "Top 3 frames of crashing thread:\n"
+            "```\n"
+            "0  fake_module  foo::bar(char* x, int y)  fake.cpp:1\n"
+            "1  fake_module  foo::bar(char* x, int y)  fake.cpp:1\n"
+            "2  fake_module  foo::bar(char* x, int y)  fake.cpp:1\n"
+            "```"
```

Comment:
I changed all the comment tests so that instead of asserting a substring in a url which is quoted which is impossible to infer anything useful about and excrutiating to verify, it's now explicit and sensible.